crates/duckdb/src/lib.rs

```diff
@@ -357,132 +257,166 @@ pub fn version() -> &'static str {
     env!("CARGO_PKG_VERSION")
 }
 
+fn to_geoarrow_record_batch(mut record_batch: RecordBatch) -> Result<RecordBatch> {
+    if let Some((index, _)) = record_batch.schema().column_with_name("geometry") {
+        let geometry_column = record_batch.remove_column(index);
+        let binary_array: GenericByteArray<GenericBinaryType<i32>> =
+            arrow::array::downcast_array(&geometry_column);
+        let wkb_array = WKBArray::new(binary_array, Default::default());
+        let geometry_array = geoarrow::io::wkb::from_wkb(
+            &wkb_array,
+            NativeType::Geometry(CoordType::Interleaved),
+            false,
+        )?;
+        let mut columns = record_batch.columns().to_vec();
+        let mut schema_builder = SchemaBuilder::from(&*record_batch.schema());
+        schema_builder.push(geometry_array.extension_field());
+        let schema = schema_builder.finish();
+        columns.push(geometry_array.to_array_ref());
+        record_batch = RecordBatch::try_new(schema.into(), columns)?;
```

Comment:
@kylebarron no rush, but if you wouldn't mind putting eyes on this bit to tell me what I'm doing wrong 🙇🏼.

Reply:
What are you trying to do?

Reply:
Take the binary column returned by https://docs.rs/duckdb/latest/duckdb/struct.Statement.html#method.query_arrow and set it up so that I can build a [Table](https://docs.rs/geoarrow/0.4.0-beta.3/geoarrow/table/struct.Table.html) from the query restuls.

Reply:
Other than the comments above, it looks fine. It's a bit verbose, but you need to pop the geometry column out of the schema and the batch, mutate each of them, and then add them back onto the schema and batch.

Reply:
Thanks Kyle!